vespa-maven-plugin/src/main/java/ai/vespa/hosted/plugin/CompileVersionMojo.java

```diff
@@ -30,14 +32,16 @@ public class CompileVersionMojo extends AbstractVespaMojo {
     protected void doExecute() throws IOException {
         Path output = Paths.get(outputFile).toAbsolutePath();
         OptionalInt allowMajor = majorVersion(new File(project.getBasedir(), "src/main/application/deployment.xml").toPath());
-        String compileVersion = controller.compileVersion(id, allowMajor);
-        if (allowMajor.isPresent()) {
-            getLog().info("Allowing major version " + allowMajor.getAsInt() + ".");
-        }
-        getLog().info("Vespa version to compile against is '" + compileVersion + "'.");
+        allowMajor.ifPresent(major -> getLog().info("Allowing only major version " + major + "."));
+
+        Version compileVersion = Version.fromString(controller.compileVersion(id, allowMajor));
+        if (compileVersion.isAfter(Vtag.currentVersion))
+            compileVersion = Vtag.currentVersion;
```

Comment:
That's what I wonder, yes ... 
Currently, we fail later, when trying to compile against a version which is newer than the parent. This happens in the package step. We want to keep that check.
We could, however, fail earlier (here) with a more specialised error message, pointing out that the parent version isn't the newest one available—the controller only considers compile versions it can find in the designated maven repository for each system.

Reply:
> We could, however, fail earlier (here) with a more specialised error message

I vote for this. If it's going to fail anyway, earlier is better.

Reply:
Agreed.